docs/tuning.md

```diff
@@ -264,6 +264,13 @@ parent RDD's number of partitions. You can pass the level of parallelism as a se
 or set the config property `spark.default.parallelism` to change the default.
 In general, we recommend 2-3 tasks per CPU core in your cluster.
 
+Sometimes you may also need to increase directory listing parallelism when job input has large number of directories,
+otherwise the process could take a very long time, especially when against object store like S3.
```

Comment:
What about remote HDFS? Since this looks like a general issue for remote storage access. Especially, in disaggregated clusters where remote storage (HDFS/S3) are used, can we generalize more like the following?
```
- especially when against object store like S3
- especially when against remote HDFS or S3 or in the disaggregated clusters
```

Reply:
It depends on how "remote" the storage is. For HDFS, depending on the use case the compute and storage can still deployed within the same region and therefore network/metadata cost is much cheaper than that from S3.

Therefore, IMO think we can stick with the S3 case as it is more characteristic. Let me know if you think otherwise.